src/Components/Web.JS/src/Platform/Circuits/DefaultReconnectDisplay.ts

```diff
@@ -2,165 +2,242 @@
-    link.addEventListener('click', () => location.reload());
+    .${this.ReconnectOverlayClassName} button:hover {
+      background-color: #3b6ea2;
+    }
 
-    const errorInstructions = this.document.createTextNode(' the page to restore functionality.');
+    .${this.ReconnectOverlayClassName} button:active {
+      background-color: #6b9ed2;
+    }
 
-    this.message.replaceChildren(errorDescription, link, errorInstructions);
-  }
+    .${this.ReconnectDialogClassName} {
+      position: relative;
+      background-color: white;
+      width: 20rem;
+      margin: 20vh auto;
+      padding: 2rem;
+      border-radius: 0.5rem;
+      box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.3);
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      gap: 1rem;
+      opacity: 0;
+      animation: components-reconnect-slideUp 1.5s cubic-bezier(.05, .89, .25, 1.02) 0.3s, components-reconnect-fadeInOpacity 0.5s ease-out 0.3s;
+      animation-fill-mode: forwards;
+      z-index: 10001;
+    }
 
-  private getLoader(): HTMLDivElement {
-    const loader = this.document.createElement('div');
-
-    const loaderStyles = [
-      'border: 0.3em solid #f3f3f3',
-      'border-top: 0.3em solid #3498db',
-      'border-radius: 50%',
-      'width: 2em',
-      'height: 2em',
-      'display: inline-block',
-    ];
-
-    loader.style.cssText = loaderStyles.join(';');
-    loader.animate([{ transform: 'rotate(0deg)' }, { transform: 'rotate(360deg)' }], {
-      duration: 2000,
-      iterations: Infinity,
-    });
-
-    return loader;
-  }
+    .${this.RejoiningAnimationClassName} {
+      display: block;
+      position: relative;
+      width: 80px;
+      height: 80px;
+    }
+
+    .${this.RejoiningAnimationClassName} div {
+      position: absolute;
+      border: 3px solid #0087ff;
+      opacity: 1;
+      border-radius: 50%;
+      animation: ${this.RejoiningAnimationClassName} 1.5s cubic-bezier(0, 0.2, 0.8, 1) infinite;
+    }
+
+    .${this.RejoiningAnimationClassName} div:nth-child(2) {
+      animation-delay: -0.5s;
+    }
+
+    @keyframes ${this.RejoiningAnimationClassName} {
+      0% {
+        top: 40px;
+        left: 40px;
+        width: 0;
+        height: 0;
+        opacity: 0;
+      }
+
+      4.9% {
+        top: 40px;
+        left: 40px;
+        width: 0;
+        height: 0;
+        opacity: 0;
+      }
+
+      5% {
+        top: 40px;
+        left: 40px;
+        width: 0;
+        height: 0;
+        opacity: 1;
+      }
+
+      100% {
+        top: 0px;
+        left: 0px;
+        width: 80px;
+        height: 80px;
+        opacity: 0;
+      }
+    }
+
+    @keyframes components-reconnect-fadeInOpacity {
+      0% {
+        opacity: 0;
+      }
+
+      100% {
+        opacity: 1;
+      }
+    }
+
+    @keyframes components-reconnect-slideUp {
+      0% {
+        transform: translateY(30px) scale(0.95);
+      }
+
+      100% {
+        transform: translateY(0);
+      }
+    }
+  `;
```

Comment:
I feel this has reached the point of what its acceptable to drop in the bundled js file. I'm wondering if there is a better way to style this UI that doesn't involve bundling a bunch of non-trivial styles on the project.

Maybe we can find a better separation of concerns by moving some of these styles to a stylesheet that we drop on the template. That will have 2 benefits:
* Clear separation of concerns.
* A starting point to customize the UI to suit the look and feel of your app.

Reply:
Would something like what we do for the loading spinner work here?

Reply:
Yeah, that's an option; it just means that existing apps won't automatically get the new styles. Maybe that's fine. We could assess that after we finalize what the UX should be, if that's okay.

Reply:
We concluded offline that we're not going to bring these changes into the template for now.